master/internal/api_tasks.go

```diff
@@ -300,6 +300,43 @@ func (a *apiServer) TaskLogs(
 	})
 }
 
+func (a *apiServer) PostTaskLogs(
+	ctx context.Context, req *apiv1.PostTaskLogsRequest,
+) (*apiv1.PostTaskLogsResponse, error) {
+	if len(req.Logs) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "len logs must be greater than 0")
+	}
+	taskID := req.Logs[0].TaskId
+
+	if err := a.canDoActionsOnTask(ctx, model.TaskID(taskID),
+		expauth.AuthZProvider.Get().CanEditExperiment); err != nil {
+		return nil, err
+	}
+
+	logs := make([]*model.TaskLog, len(req.Logs))
+	for i := range req.Logs {
+		if req.Logs[i].Id != nil {
+			return nil, status.Errorf(codes.InvalidArgument,
+				"ID must be nil on logs got %d instead", *req.Logs[i].Id)
+		}
+		if req.Logs[i].TaskId != taskID {
+			// There isn't a hard reason for this requirement other than we would have to RBAC
+			// against all provided taskIDs. This usecase seems pretty unlikely.
```

Comment:
question (non-blocking): Do you know of any cases where this could occur? Or is this preventive and/or for checking assumptions?

Reply:
Not from our ```enrich_task_logs.py```, but you could technically just call this endpoint from user code and pass in whatever arguments you want causing something like this to happen.